core/templates/domain/collection/collection-playthrough.model.spec.ts

```diff
@@ -0,0 +1,173 @@
+    const collectionPlaythroughBackendDictCase1 = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1']
+    };
+    const collectionPlaythroughBackendDictCase2 = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: []
+    };
+    const collectionPlaythroughBackendDictCase3 = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1', 'exp_id2']
+    };
+    const collectionPlaythroughObjectCase1 = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDictCase1);
+    const collectionPlaythroughObjectCase2 = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDictCase2);
+    const collectionPlaythroughObjectCase3 = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDictCase3);
+    expect(
+      collectionPlaythroughObjectCase1._completedExplorationIds
+    ).toEqual(['exp_id1']);
+    expect(
+      collectionPlaythroughObjectCase2._completedExplorationIds
+    ).toEqual([]);
+    expect(
+      collectionPlaythroughObjectCase3._completedExplorationIds
+    ).toEqual(['exp_id1', 'exp_id2']);
+  });
+
+  it('should get the completed exploration node count', () => {
+    const collectionPlaythroughBackendDictCase1 = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1']
+    };
+    const collectionPlaythroughBackendDictCase2 = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: []
+    };
+    const collectionPlaythroughBackendDictCase3 = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: ['exp_id1', 'exp_id2']
+    };
+    const collectionPlaythroughObjectCase1 = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDictCase1);
+    const collectionPlaythroughObjectCase2 = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDictCase2);
+    const collectionPlaythroughObjectCase3 = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDictCase3);
+    expect(collectionPlaythroughObjectCase1.getCompletedExplorationNodeCount())
+      .toEqual(1);
+    expect(collectionPlaythroughObjectCase2.getCompletedExplorationNodeCount())
+      .toEqual(0);
+    expect(collectionPlaythroughObjectCase3.getCompletedExplorationNodeCount())
+      .toEqual(2);
+  });
+
+  it('should return collection is not started when no exploration is' +
+    'completed',
+  () => {
+    const collectionPlaythroughBackendDict = {
+      next_exploration_id: 'exp_id0',
+      completed_exploration_ids: []
+    };
+    const collectionPlaythroughObject = CollectionPlaythrough
+      .createFromBackendObject(collectionPlaythroughBackendDict);
+    expect(collectionPlaythroughObject.hasStartedCollection())
+      .toEqual(false);
+  }
+  );
+
+  it(
+    'should return collection is started when any exploration is' +
+      'completed',
```

Comment:
For consistency, can you not break at the parentheses here? Ditto for below.

Reply:
Done